assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -91,45 +87,53 @@ const SettingsSection = props => {
 			props
 		);
 	};
+	let columns;
+	if ( fields.length % 3 === 0 ) {
+		columns = 3;
+	} else if ( fields.length % 2 === 0 ) {
+		columns = 2;
+	} else {
+		columns = 1;
+	}
 	return (
 		<ActionCard
 			isMedium
 			disabled={ disabled }
 			title={ title }
 			description={ description }
 			toggleChecked={ active }
-			hasGreyHeader={ true }
+			hasGreyHeader={ active || hasGreyHeader }
 			toggleOnChange={ active !== null ? value => onUpdate( { active: value } ) : null }
+			actionContent={
+				( active || hasGreyHeader ) &&
```

Comment:
It is not clear to me why the save button requires a grey header or a "toggleable" setting. This obligates every setting that is not toggleable to determine a style property for the functionality to work. Not having the grey header by default in such a case also results in an unexpected style:

<img width="1078" alt="image" src="https://user-images.githubusercontent.com/820752/155745436-df483be5-1dc4-4a88-a49c-941ad7c6de89.png">

Reply:
In your screenshot, you should have `hasGreyHeader`

Reply:
You can see that [line 124 uses `activer || active === null`](https://github.com/Automattic/newspack-plugin/blob/a3649c8629c6a6ae81421bf7f4137a62e4036756/assets/components/src/plugin-settings/SettingsSection.js#L124) to determine if settings should be displayed, which is the same statement that should determine a grey header for this component.

Reply:
Is there ever a use case for the non-grey header style in @miguelpeixe's screenshot?